src/fifo.ts

```diff
@@ -57,41 +57,32 @@ export interface FIFOOptions {
    * When the queue reaches this size, it will be split into head/tail parts
    */
   splitLimit?: number
-
-  /**
-   * If true, `size` will be the number of items in the queue. If false, all
-   * values will be interpreted as Uint8Arrays and `size` will be the total
-   * number of bytes in the queue.
-   */
-  objectMode?: boolean
 }
 
 export class FIFO<T> {
   public size: number
   private readonly hwm: number
   private head: FixedFIFO<T>
   private tail: FixedFIFO<T>
-  private readonly objectMode: boolean
 
   constructor (options: FIFOOptions = {}) {
     this.hwm = options.splitLimit ?? 16
     this.head = new FixedFIFO<T>(this.hwm)
     this.tail = this.head
     this.size = 0
-    this.objectMode = Boolean(options.objectMode)
+  }
+
+  calculateSize (obj: any): number {
```

Comment:
```suggestion
  calculateSize (obj: unknown): number {
```

Reply:
If I make this change, TS does not allow me to access the property `.byteLength` because the type is `unknown`.

Reply:
It should allow with https://github.com/alanshaw/it-pushable/pull/30#discussion_r935313042 and this would ensure runtime safety

Reply:
No, then I get `Property 'byteLength' does not exist on type 'never'.`

Reply:
This seems to work:

```js
calculateSize (obj: T | { byteLength: number }): number {
  if (obj != null && 'byteLength' in obj && obj.byteLength != null) {
    return obj.byteLength
  }

  return 1
}
```

Reply:
Ha, no it doesn't..

Reply:
interesting, I thought it was fixed in the recent version of typescript. Maybe it wasn't.
That works too!
I used to use this helper for that:
```
  function hasOwnProperty<O extends {}, P extends PropertyKey> (obj: O, prop: P): obj is O & Record<P, unknown> {
    return obj.hasOwnProperty(prop)
  }
```

Reply:
> Ha, no it doesn't..

You are missing `typeof obj == 'object'` 😄

Reply:
I feel like we are nitpicking too much, nobody is using this class directly so I think original check if fine 😄

Reply:
Yes, if I add `typeof obj == 'object'` then it thinks the type is `"object"` which doesn't have a `"byteLength"` property, hence you get the error `Property 'byteLength' does not exist on type 'object'.`

Reply:
Oops, my comment arrived after yours.  Fair point though, let's go with this and we can change if needed.